### PR TITLE
Implement competition select page 

### PR DESCRIPTION
**Pull request by lin8231**
* UI is implemented 
* All displayed data is pulled from the back end 
* Competition cards are dynamically rendered given number of objects received from backend
* The user param used for querying is currently hard coded to 1 
* Routing now works properly between sign in/compselect/in-game screens
* Create competition button currently does nothing - no implementation yet 
* Competition cards can be horizontally scrolled using mouse wheel 
* Selected competition is stored in local storage so when user refreshes the compId persists 
* There was an extra package-lock.json (on top of the existing one) for some reason, now removed (it was also somehow not being gitignored).

### CODE REVIEW

File: project/frontend/src/components/CompetitionCard/CompetitionCard.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React, { useEffect } from "react";
+import classes from "./CompetitionCard.module.css";
+import LinkButton from "../UI/LinkButton/LinkButton";
+
+const CompetitionCard = (props) => {
+  const getNumberWithOrdinal = (n) => {
+    var s = ["th", "st", "nd", "rd"],
+      v = n % 100;
+    return n + (s[(v - 20) % 10] || s[v] || s[0]);
```
Comment (roonil-wazlib) [inline project/frontend/src/components/CompetitionCard/CompetitionCard.js:9]: Might want a comment explaining this black magic

**Review by roonil-wazlib — CHANGES_REQUESTED** [review]

**Comment by roonil-wazlib** [comment]
Can you fix the conflicts and confirm everything works before merging?

File: project/frontend/src/components/CompetitionCard/CompetitionCard.js (hunk 1 above)
Comment (lin8231) [reply project/frontend/src/components/CompetitionCard/CompetitionCard.js:9]: True lmao

Comment (lin8231) [reply project/frontend/src/components/CompetitionCard/CompetitionCard.js:14]: Done

**Comment by lin8231** [comment]
> Can you fix the conflicts and confirm everything works before merging?

Done

**Review by roonil-wazlib — APPROVED** [review]
LGTM

**Review by leonnemets — APPROVED** [review]

**Pull request merged**